home/jobs/BIOFORMATS-test-repo/config.xml

```diff
@@ -0,0 +1,44 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<flow-definition plugin="workflow-job@2.11.1">
+  <actions/>
+  <description></description>
+  <keepDependencies>false</keepDependencies>
+  <properties>
+    <org.jenkinsci.plugins.workflow.job.properties.PipelineTriggersJobProperty>
+      <triggers/>
+    </org.jenkinsci.plugins.workflow.job.properties.PipelineTriggersJobProperty>
+  </properties>
+  <definition class="org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition" plugin="workflow-cps@2.36.1">
+    <script>def transformIntoStep(inputString) {
+    return {
+        //stage(inputString) {
+            //println inputString
+            build job: &apos;BIOFORMATS-test-folder&apos;, parameters: [
+            [$class: &apos;StringParameterValue&apos;, name: &apos;FORMAT_NAME&apos;, value: inputString]]
+        //}
+    }
+}
+
+def stepsForParallel = []
+
+node(&quot;docker&quot;) {
+    stage(&apos;List&apos;) {
+        def data_dir = System.getenv()['REPO_CURATED']
+        def cmd = &quot;find ${data_dir} -mindepth 1 -maxdepth 1 -type d &quot;
+        list = sh(returnStdout: true, script: cmd).trim()
+        def formats = list.split()
+        stepsForParallel = formats.collectEntries {
+            [&quot;handling ${it}&quot; : transformIntoStep(it.substring(it.lastIndexOf(&apos;/&apos;) + 1))]
+        }
+    }
+    
+}
+
+stage(&apos;Run&apos;) {
+    parallel stepsForParallel
+}</script>
+    <sandbox>false</sandbox>
```

Comment:
For me the first execution failed with 

```
org.jenkinsci.plugins.scriptsecurity.scripts.UnapprovedUsageException: script not yet approved for use
	at org.jenkinsci.plugins.scriptsecurity.scripts.ScriptApproval.using(ScriptApproval.java:466)
	at org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition.create(CpsFlowDefinition.java:117)
	at org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition.create(CpsFlowDefinition.java:64)
	at org.jenkinsci.plugins.workflow.job.WorkflowRun.run(WorkflowRun.java:271)
	at hudson.model.ResourceController.execute(ResourceController.java:97)
	at hudson.model.Executor.run(Executor.java:429)
```

Should we sandbox this one to improve the deployment experience?

Reply:
Ah, I hadn't noticed an XML-way of fixing that. Sure.